kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyReconciler.java

```diff
@@ -411,7 +413,11 @@ private static ConfigurationFragment<Optional<TrustProvider>> buildTrustProvider
     @Override
     public UpdateControl<KafkaProxy> reconcile(KafkaProxy primary,
                                                Context<KafkaProxy> context) {
-        var uc = UpdateControl.patchStatus(statusFactory.newTrueConditionStatusPatch(primary, Condition.Type.Ready, ""));
+        MetadataChecksumGenerator checksumGenerator = context.managedWorkflowAndDependentResourceContext()
+                .get(MetadataChecksumGenerator.CHECKSUM_CONTEXT_KEY, MetadataChecksumGenerator.class)
+                .orElse(new Crc32ChecksumGenerator());
+        checksumGenerator.appendMetadata(primary);
```

Comment:
I wonder if we could create a single checksum early on in the initialize, considering all referents and the primary, and then use it as both the checksum on the template and here as an annotation.

So refactor so we use the same checksum in both spots.

Reply:
I sort of think they are different things. This one is the chcksum of just the proxy (and should be fed into the one in the template, thats why this PR is still in draft), the cheksum in the deployment is meant to map to a specific state of the config map.

You could be right however we should calculate them in init context.